src/Box.tsx

```diff
@@ -30,50 +29,59 @@ import {
   zIndex,
 } from './utils';
 
-import { BorderStyleProperty } from 'csstype';
-import { BorderRadius, BorderWidth, BoxDimension, BoxShadow, FontSize, FontWeight, FullSpace } from './types';
+import {
+  IBorderRadius,
+  IBorderWidth,
+  IBoxDimension,
+  IFontSize,
+  IFontWeight,
+  IFullSpace,
+  IShadows,
+  ValueOf,
+} from './types';
 
 export interface IBoxProps {
   className?: string;
   css?: Object; // a valid javascript style object
+  as?: any; // TODO type string || jsx element
 
   fg?: string; // should ALWAYS be a theme color path to avoid hardcoding in values
   bg?: string; // should ALWAYS be a theme color path to avoid hardcoding in values
 
-  text?: FontSize;
+  text?: ValueOf<IFontSize>;
   align?: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit'; // textAlign
-  weight?: FontWeight;
-
-  m?: FullSpace;
-  mt?: FullSpace;
-  mr?: FullSpace;
-  mb?: FullSpace;
-  ml?: FullSpace;
-  mx?: FullSpace;
-  my?: FullSpace;
-  p?: FullSpace;
-  pt?: FullSpace;
-  pr?: FullSpace;
-  pb?: FullSpace;
-  pl?: FullSpace;
-  px?: FullSpace;
-  py?: FullSpace;
-
-  height?: BoxDimension | string | number;
-  maxHeight?: BoxDimension | string | number;
-  minHeight?: BoxDimension | string | number;
-  width?: BoxDimension | string | number;
-  maxWidth?: BoxDimension | string | number;
-  minWidth?: BoxDimension | string | number;
-
-  border?: BorderWidth;
-  borderTop?: BorderWidth;
-  borderLeft?: BorderWidth;
-  borderRight?: BorderWidth;
-  borderBottom?: BorderWidth;
+  weight?: ValueOf<IFontWeight>;
+
+  m?: ValueOf<IFullSpace>;
+  mt?: ValueOf<IFullSpace>;
+  mr?: ValueOf<IFullSpace>;
+  mb?: ValueOf<IFullSpace>;
+  ml?: ValueOf<IFullSpace>;
+  mx?: ValueOf<IFullSpace>;
+  my?: ValueOf<IFullSpace>;
+  p?: ValueOf<IFullSpace>;
+  pt?: ValueOf<IFullSpace>;
+  pr?: ValueOf<IFullSpace>;
+  pb?: ValueOf<IFullSpace>;
+  pl?: ValueOf<IFullSpace>;
+  px?: ValueOf<IFullSpace>;
+  py?: ValueOf<IFullSpace>;
+
+  height?: ValueOf<IBoxDimension> | string | number;
+  maxHeight?: ValueOf<IBoxDimension> | string | number;
+  minHeight?: ValueOf<IBoxDimension> | string | number;
+  width?: ValueOf<IBoxDimension> | string | number;
+  maxWidth?: ValueOf<IBoxDimension> | string | number;
```

Comment:
Same as above SpacingVal comment but for this situation. Same anywhere else that we repeat the same typing more than like 5 times or something.